imap_processing/spice/repoint.py

```diff
@@ -0,0 +1,118 @@
+"""Functions for retrieving repointing table data."""
+
+import logging
+import os
+from pathlib import Path
+from typing import Union
+
+import numpy as np
+import pandas as pd
+from numpy import typing as npt
+
+logger = logging.getLogger(__name__)
+
+
+def get_repoint_data() -> pd.DataFrame:
+    """
+    Read repointing file using environment variable and return as dataframe.
+
+    REPOINT_DATA_FILEPATH environment variable should point to a local
+    file where the repointing csv file is located.
+
+    Returns
+    -------
+    repoint_df : pd.DataFrame
+        The repointing csv loaded into a pandas dataframe. The dataframe will
+        contain the following columns:
+            - `repoint_start_time`: Starting MET time of each repoint maneuver.
+            - `repoint_end_time`: Ending MET time of each repoint maneuver.
+            - `repoint_id`: Unique ID number of each repoint maneuver.
+    """
+    repoint_data_filepath = os.getenv("REPOINT_DATA_FILEPATH")
+    if repoint_data_filepath is not None:
+        path_to_spin_file = Path(repoint_data_filepath)
+    else:
+        # Handle the case where the environment variable is not set
+        raise ValueError("REPOINT_DATA_FILEPATH environment variable is not set.")
+
+    logger.info(f"Reading repointing data from {path_to_spin_file}")
+    repoint_df = pd.read_csv(path_to_spin_file, comment="#")
+
+    return repoint_df
+
+
+def interpolate_repoint_data(
+    query_met_times: Union[float, npt.NDArray],
+) -> pd.DataFrame:
+    """
+    Interpolate repointing data to the queried MET times.
+
+    Query times that are between a pointing end time and the next pointing start
+    time will cause an exception to be raised.
+
+    Parameters
+    ----------
+    query_met_times : float or np.ndarray
+        Query times in Mission Elapsed Time (MET).
+
+    Returns
+    -------
+    repoint_df : pandas.DataFrame
+        Repoint table data with the interpolated such that there is one row
+        for each of the queried MET times. Output columns are:
+            - `repoint_start_time`
+            - `repoint_end_time`
+            - `repoint_id`
+            - `repoint_in_progress`
+
+    Raises
+    ------
+    ValueError : If any of the query_met_times are outside the set of
+    [repoint_start_time, repoint_end_time] time ranges defined in the repoint
+    table.
+    """
+    repoint_df = get_repoint_data()
+
+    # Ensure query_met_times is an array
+    query_met_times = np.asarray(query_met_times)
+    is_scalar = query_met_times.ndim == 0
+    if is_scalar:
+        # Force scalar to array because np.asarray() will not
+        # convert scalar to array
```

Comment:
Can you use this directly above instead of np.asarray?
`query_met_times = np.atleast_1d(query_met_times)`